### ISSUE TITLE
The ability of exporting Resonite packages

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1441]
### Is your feature request related to a problem? Please describe.

Right now the only way to share content outside of the game is to share public folder links and if you wanted to sell your content on marketplaces like gumroad. It's really difficult to give your customers what they paid for without giving them a public folder link or meeting the end game to hand them the avatar

### Describe the solution you'd like

A way of exporting our content with all assets and functionality in a nice neat package file that will be future proof and will automatically update so nothing will break if the game gets updates in the future

### Describe alternatives you've considered

Using things like .lz4bson but these types of files are extremely volatile if the asset that it's referencing is deleted or changed the links that it's referencing will be completely invalid a Resonite package would avoid this

### Requesters

@epicEaston197 and way too many others to list here

**Comment by 989onan** [comment]
I agree on this one. A resonite specific package that is less """""flakey""""" and user friendly than a brson would be nice

**Comment by Frooxius** [comment]
This is something we can look into at some point.

Just a note here, brzon, 7zbson and so on were never designed or intended for object exhange like this. They are definitely very fragile for this kind of use and we strongly discourage their usage for this purpose.

**Comment by 989onan** [comment]
yeah I figured as much. It's why I try to stay away from them as much as possible. Is why I agree with the title of the issue that we should have a system that can pack Ids and their respective files into a file format to give to others, rather than such a """""flakey""""" format

This is just a suggestion and probably not how you'll implement it, but:
It would be nice to allow uploading of assets via the file format, and if the cloud finds a hash of an asset with the same data, it will use that instead of uploading it as a new asset
But then again there are better implementations like the original creator keeping it stored on their account and so on

Gets the ideas flowing for sure

**Comment by Frooxius** [comment]
It's not that the format is "flakey", but it's being used for something that it's not intended, nor designed for. 

To give analogy, it's like using clothes tied up together into a rope for rock climbing. Is that going to be flakey? Yes, but it's not that the clothes are flakey - you're using them for something they're not meant for.


The cloud already handles deduplication via hashes, but that doesn't really matter here much - it's not something the format has to care about on itself - its assets would be extracted from it and then follow the normal pipeline.

**Comment by ModernBalloonie** [comment]
My main problem is that I sell avatars, so having an easy way to drag it into the game would be really nice, because currently without it, I have to ask people who buy it to message me, and i'd go over there and give the avatars to them personally. I was thinking of including a brson to easily just get the avatar without having to do much besides put avatar protection on it.
